metadata/src/main/java/com/redhat/lightblue/metadata/ValueGenerator.java

```diff
@@ -0,0 +1,42 @@
+package com.redhat.lightblue.metadata;
+
+import java.io.Serializable;
+import java.util.Properties;
+
+/**
+ * Represents value generator definition located in entityInfo.
+ *
+ * @author mpatercz
+ *
+ */
+public class ValueGenerator implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    public enum ValueGeneratorType {
+        IntSequance;
+    }
+
```

Comment:
Disregard my previous question.

I'm having a problem grasping how a new value type can be used with generators. Generators generate values of existing types. For numeric sequences it will be BigInteger, so BigInteger.toJson(generatedValue).